src/librustc/infer/error_reporting/nice_region_error/placeholder_error.rs

```diff
@@ -311,13 +314,46 @@ impl NiceRegionError<'me, 'gcx, 'tcx> {
         sup_placeholder: Option<ty::Region<'tcx>>,
         has_sub: Option<usize>,
         has_sup: Option<usize>,
-        expected_trait_ref: ty::TraitRef<'_>,
-        actual_trait_ref: ty::TraitRef<'_>,
+        expected_trait_ref: ty::TraitRef<'tcx>,
+        actual_trait_ref: ty::TraitRef<'tcx>,
         vid: Option<ty::Region<'tcx>>,
         expected_has_vid: Option<usize>,
         actual_has_vid: Option<usize>,
         any_self_ty_has_vid: bool,
     ) {
+        // HACK(eddyb) maybe move this in a more central location.
+        #[derive(Copy, Clone)]
+        struct Highlighted<'a, 'gcx, 'tcx, T> {
```

Comment:
It *does* seem a bit odd for this struct to be private to this function, when the region highlighting machinery *itself* all likes in the generic `ty::print` module. We could move it there I guess, but I'm also ok to hold off a bit.

Reply:
I would want to move it out of `ty::print::pretty`, but maybe not in this PR.